app/Http/Controllers/TransportController.php

```diff
@@ -577,22 +613,25 @@ public static function getPolylineHash(string $polyline): PolyLine {
     /**
      * Get the latest TrainStations the user is arrived.
      * @param User $user
-     * @param int $maxCount
+     * @param int $limit
      * @return Collection
      */
-    public static function getLatestArrivals(User $user, int $maxCount = 5): Collection {
-        $user->loadMissing(['statuses.trainCheckIn.Destination']);
-        return $user->statuses
-            ->map(function($status) {
-                return $status->trainCheckIn;
-            })
-            ->sortByDesc('arrival')
-            ->map(function($checkIn) {
-                return $checkIn->Destination;
-            })->groupBy('ibnr')
-            ->map(function($trainStations) {
-                return $trainStations->first();
-            })->take($maxCount);
+    public static function getLatestArrivals(User $user, int $limit = 5): Collection {
+        return $user->statuses()
+                    ->join('train_checkins', 'train_checkins.status_id', '=', 'statuses.id')
+                    ->join('train_stations', 'train_stations.ibnr', '=', 'train_checkins.destination')
+                    ->orderByDesc('train_checkins.arrival')
+                    ->groupBy('train_stations.name')
+                    ->select([
+                                 'train_stations.id',
+                                 'train_stations.ibnr',
+                                 'train_stations.rilIdentifier',
+                                 'train_stations.name',
+                                 'train_stations.latitude',
+                                 'train_stations.longitude'
+                             ])
+                    ->limit($limit)
+                    ->get();
```

Comment:
Error in this query.

![image](https://user-images.githubusercontent.com/4103693/126866097-f082fe46-ffb2-41a6-bb42-3c7bee2a7ef5.png)